lib/rihanna/job.ex

```diff
@@ -252,7 +275,7 @@ defmodule Rihanna.Job do
           WHERE NOT (id = ANY($3))
           AND (due_at IS NULL OR due_at <= now())
           AND failed_at IS NULL
-          ORDER BY enqueued_at, j.id
+          ORDER BY priority, enqueued_at, j.id
```

Comment:
Have you verified it still uses the index correctly? This is a hot path and we need to be extremely sure it's optimised.

Reply:
I wasn't sure how to make an automated test for this, but I looked at `EXPLAIN` and the queries I tried looked good.

Reply:
```
Limit  (cost=1723.71..1723.93 rows=6 width=97)
  CTE jobs
    ->  Recursive Union  (cost=0.29..1723.71 rows=11 width=97)
          ->  Subquery Scan on t1  (cost=0.29..63.33 rows=1 width=97)
                ->  Limit  (cost=0.29..63.31 rows=1 width=174)
                      ->  LockRows  (cost=0.29..1260.85 rows=20 width=174)
                            ->  Index Scan using rihanna_jobs_priority_enqueued_at_id on rihanna_jobs j  (cost=0.29..1260.64 rows=20 width=174)
                                  Filter: ((failed_at IS NULL) AND (id <> ALL ('{1}'::integer[])) AND ((due_at IS NULL) OR (due_at <= now())))
          ->  Subquery Scan on t1_1  (cost=0.00..166.02 rows=1 width=97)
                ->  Limit  (cost=0.00..166.00 rows=1 width=32)
                      ->  WorkTable Scan on jobs jobs_1  (cost=0.00..1660.04 rows=10 width=32)
                            Filter: (id IS NOT NULL)
  ->  CTE Scan on jobs  (cost=0.00..0.22 rows=6 width=97)
        Filter: locked
```

Reply:
I think the important part is that it does the `Index Scan using rihanna_jobs_priority_enqueued_at_id`.

The rest looks the same as on my production instance.

Reply:
Yeah looks good to me.